presto-hive/src/main/java/io/prestosql/plugin/hive/HivePageSource.java

```diff
@@ -163,7 +163,9 @@ public HivePageSource(
                         .orElse(ImmutableList.of());
                 HiveType fromType = columnMapping.getBaseTypeCoercionFrom().get().getHiveTypeForDereferences(dereferenceIndices).get();
                 HiveType toType = columnMapping.getHiveColumnHandle().getHiveType();
-                coercers.add(Optional.of(createCoercer(typeManager, fromType, toType)));
+                if (!fromType.equals(toType)) {
```

Comment:
The `coercers` list is expected to be the same size as `columnMapping`. I think that'll require an empty coercer to be added here.